### PR TITLE
Update References

### LINKED ISSUE #1048: Update References chapter
Going through the model (then the spec), it seems we refer to different things not mentioned in the References chapter.

For example, it has only 5 of the 11 referenced RFCs.

I'll add the missing references.

### PR DESCRIPTION
**Pull request by zvr**
This adds a number of resources referenced in the spec chapters and model.

Closes #1048

Signed-off-by: Alexios Zavras <github@zvr.gr>

### CODE REVIEW

File: docs/references.md [hunk 1]
```diff
@@ -17,47 +17,137 @@ edition of the referenced document (including any amendments) applies.
 The MITRE Corporation,
 [https://cpe.mitre.org/files/cpe-specification_2.2.pdf](https://cpe.mitre.org/files/cpe-specification_2.2.pdf)
 
-NIST IR 7695, *Common Platform Enumeration: Naming Specification Version 2.3*,
+*Common Platform Enumeration: Naming Specification Version 2.3*, NIST IR 7695,
```
Comment (kestewart) [inline docs/references.md:20]:
```suggestion
*Common Platform Enumeration(CPE): Naming Specification Version 2.3*, NIST IR 7695,
```

File: docs/references.md [hunk 2]
```diff
@@ -17,47 +17,137 @@ edition of the referenced document (including any amendments) applies.
...
 NIST,
 [https://csrc.nist.gov/pubs/ir/7695/final](https://csrc.nist.gov/pubs/ir/7695/final)
 
-*npm-package.json*, npm Inc.,
-[https://docs.npmjs.com/files/package.json](https://docs.npmjs.com/files/package.json)
+*Common Vulnerability Scoring System v3.0: Specification Document*,
```
Comment (kestewart) [inline docs/references.md:24]:
```suggestion
*Common Vulnerability Scoring System v3.0 (CVSS v3.0): Specification Document*,
```

File: docs/references.md [hunk 3]
```diff
@@ -17,47 +17,137 @@ edition of the referenced document (including any amendments) applies.
...
+Forum of Incident Response and Security Teams, Inc (FIRST),
+[https://www.first.org/cvss/v3.0/specification-document](https://www.first.org/cvss/v3.0/specification-document)
 
-*NuGet documentation*, Microsoft,
-[https://docs.nuget.org/](https://docs.nuget.org/)
+*Common Vulnerability Scoring System v3.1: Specification Document*,
```
Comment (kestewart) [inline docs/references.md:28]:
```suggestion
*Common Vulnerability Scoring System v3.1 (CVSS v3.1): Specification Document*,
```

File: docs/references.md [hunk 4]
```diff
@@ -17,47 +17,137 @@ edition of the referenced document (including any amendments) applies.
...
+Forum of Incident Response and Security Teams, Inc (FIRST),
+[https://www.first.org/cvss/v3.1/specification-document](https://www.first.org/cvss/v3.1/specification-document).
 
-POSIX.1-2017 *The Open Group Base Specifications Issue 7*, 2018 edition,
-IEEE/Open Group,
-[https://pubs.opengroup.org/onlinepubs/9699919799/](https://pubs.opengroup.org/onlinepubs/9699919799/)
+*Common Vulnerability Scoring System version 4.0: Specification Document*,
```
Comment (kestewart) [inline docs/references.md:32]:
```suggestion
*Common Vulnerability Scoring System version 4.0 (CVSS v4.0): Specification Document*,
```

File: docs/references.md [hunk 5]
```diff
@@ -17,47 +17,137 @@ edition of the referenced document (including any amendments) applies.
...
+*Semantic Versioning 2.0.0*,
+Tom Preston-Werner and SemVer contributors,
+[https://semver.org](https://semver.org).
+
+*SLSA Provenance v0.2*, The Linux Foundation,
+[https://slsa.dev/provenance/v0.2](https://slsa.dev/provenance/v0.2).
+
+SoftWare Heritage persistent IDentifiers (SWHIDs), in
+Draft International Standard 
+*ISO/IEC DIS 18670 Information technology — SoftWare Hash IDentifier (SWHID) Specification V1.2*,
```
Comment (kestewart) [inline docs/references.md:149]:
```suggestion
*ISO/IEC DIS 18670 Information technology — SoftWare Hash IDentifier (SWHID) Specification V1.2*(https://www.iso.org/standard/89985.html),
```

**Review by kestewart — CHANGES_REQUESTED** [review]
Please consider applying the fixes to help standardize the references.

**Comment by zvr** [comment]
I don't mind adding "CPE" and "CVSS", but then we won't be using the official titles any more...

File: docs/references.md (hunk 1 above)
Comment (bact) [reply docs/references.md:20]: From @kestewart suggestion, there should be a space before `(CPE)`.

**Review by bact — APPROVED** [review]
I'm good with this.

**Comment by bobmartin3000** [comment]
I'm good with this approach even though the titles are changed.

File: docs/references.md (hunk 5 above)
Comment (goneall) [reply docs/references.md:149]: [https://www.iso.org/standard/89985.html](https://www.iso.org/standard/89985.html)

**Review by bobmartin3000 — APPROVED** [review]
LGTM

File: docs/references.md (hunk 1 above)
Comment (bact) [reply docs/references.md]: I can see it fixed now

**Review by kestewart — APPROVED** [review]
Looks good now.  Thanks!

**Review by goneall — APPROVED** [review]
LGTM

**Comment by bact** [comment]
@goneall The validate failed from an issue unrelated to this PR.
Noted here https://github.com/spdx/spdx-3-model/issues/837 - to be fixed in another PR.

**Review by kestewart — APPROVED** [review]
Please consider applying the fixes to help standardize the references.

**Pull request merged**